### PR TITLE
Fix inability to toggle V-Sync dynamically

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6229

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @oleg-derevenetz ,

Outside the scope of this PR: I am thinking that for V-Sync mode we might need to issue `SDL_RenderPresent` calls in a separate thread or event skip if queue is too big. It should solve the problem with lagging cursor. What do you think?

**Comment by oleg-derevenetz** [comment]
> I am thinking that for V-Sync mode we might need to issue SDL_RenderPresent calls in a separate thread or event skip if a frame queue is too big. It should solve the problem with lagging cursor. What do you think?

I don't know is it safe to call `SDL_RenderPresent` from the separate thread. There is nothing said in the documentation regarding this.

**Review by oleg-derevenetz — APPROVED** [review]

File: src/engine/screen.cpp [hunk 1]
```diff
@@ -1194,6 +1169,59 @@ namespace
...
+                ERROR_LOG( "Failed to create a window renderer of " << width_ << " x " << height_ << " size. The error: " << SDL_GetError() )
+                clear();
+                return false;
+            }
+
+            if ( SDL_SetHint( SDL_HINT_RENDER_SCALE_QUALITY, "linear" ) == SDL_FALSE ) {
+                ERROR_LOG( "Failed to set a linear scale hint for rendering." )
+            }
+
+            // Setting this hint prevents the window to regain focus after loosing it in fullscreen mode.
```
Comment (zenseii) [inline src/engine/screen.cpp:1198]:
```suggestion
            // Setting this hint prevents the window to regain focus after losing it in fullscreen mode.
```

**Review by zenseii — CHANGES_REQUESTED** [review]
Hi, @ihhub . LGTM. I only noticed one very minor typo.

**Review by zenseii — APPROVED** [review]

**Pull request merged**